matriculation-plugin/src/main/java/fi/otavanopisto/muikku/plugins/matriculation/MatriculationPermissions.java

```diff
@@ -0,0 +1,60 @@
+package fi.otavanopisto.muikku.plugins.matriculation;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import fi.otavanopisto.muikku.model.users.EnvironmentRoleArchetype;
+import fi.otavanopisto.muikku.model.workspace.WorkspaceRoleArchetype;
+import fi.otavanopisto.muikku.security.AbstractMuikkuPermissionCollection;
+import fi.otavanopisto.muikku.security.DefaultEnvironmentPermissionRoles;
+import fi.otavanopisto.muikku.security.MuikkuPermissionCollection;
+import fi.otavanopisto.muikku.security.PermissionScope;
+import fi.otavanopisto.security.Scope;
+
+@ApplicationScoped
+public class MatriculationPermissions extends AbstractMuikkuPermissionCollection implements MuikkuPermissionCollection {
+
+  @Scope (PermissionScope.ENVIRONMENT)
+  @DefaultEnvironmentPermissionRoles ( { EnvironmentRoleArchetype.ADMINISTRATOR, EnvironmentRoleArchetype.MANAGER, EnvironmentRoleArchetype.STUDY_PROGRAMME_LEADER, EnvironmentRoleArchetype.TEACHER, EnvironmentRoleArchetype.STUDENT } )
+  public static final String MATRICULATION_GET_EXAM = "MATRICULATION_GET_EXAM";
+
+  @Scope (PermissionScope.ENVIRONMENT)
+  @DefaultEnvironmentPermissionRoles ( { EnvironmentRoleArchetype.ADMINISTRATOR, EnvironmentRoleArchetype.MANAGER, EnvironmentRoleArchetype.STUDY_PROGRAMME_LEADER, EnvironmentRoleArchetype.TEACHER, EnvironmentRoleArchetype.STUDENT } )
```

Comment:
Since the endpoint using this permission now checks that initial data belongs to the currently logged in user, I assume the endpoint is only meant for students. Thus...

@DefaultEnvironmentPermissionRoles ( { EnvironmentRoleArchetype.STUDENT } )

...should suffice.

However, this would also mean that USERID in endpoint @Path("/initialData/{USERID}") is irrelevant. If the endpoint is also meant to be used by staff to request initial data of a student, then the logged in permission check in the endpoint code renders such functionality impossible.

Reply:
Providing the user id explicitly and using the session only for permission checks reduces the number of hidden variables (the logged in user in this case) and makes the system easier to reason about and possible bugs more visible, I disagree about removing the USERID parameter, even though it's redundant, because if the endpoint is later expanded to also allow teachers and administrators, there's less changes to client side code.

Good point about the permission, I'll make it more restricted at ounce.